tensorboard/webapp/metrics/data_source/metrics_data_source.ts

```diff
@@ -266,7 +266,34 @@ export class TBMetricsDataSource implements MetricsDataSource {
       );
   }
 
-  imageUrl(imageId: ImageId) {
+  imageUrl(imageId: ImageId): string {
     return `${HTTP_PATH_PREFIX}/imageData?imageId=${imageId}`;
   }
+
+  downloadUrl(
+    pluginId: PluginType,
+    tag: string,
+    runId: string,
+    downloadType: 'json' | 'csv'
+  ): string {
+    const {run, experimentId} = parseRunId(runId);
+    let pluginAndRoute: string;
+    switch (pluginId) {
+      case PluginType.SCALARS:
+        pluginAndRoute = 'scalars/scalars';
+        break;
+      default:
+        throw new Error(
+          `Not implemented: downloadUrl for ${pluginId} is not implemented yet`
+        );
+    }
+
+    if (!experimentId) {
+      throw new Error(
+        'experimentId is empty; it is required to form downloadUrl.'
```

Comment:
I was playing around and got this error when I deselected runs:
https://screencast.googleplex.com/cast/NjA4Mjk1OTEwODk5NzEyMHwzNmUwNDlkOC0wNg

Reply:
Thanks. That had to do with the default value option on the `<select>`. Wrote a test to catch that.